src/vs/workbench/api/common/extHostExtensionService.ts

```diff
@@ -598,15 +598,22 @@ export abstract class AbstractExtHostExtensionService extends Disposable impleme
 		// startup is considered finished
 		this._mainThreadExtensionsProxy.$setPerformanceMarks(performance.getMarks());
 
-		for (const desc of this._myRegistry.getAllExtensionDescriptions()) {
-			if (desc.activationEvents) {
-				for (const activationEvent of desc.activationEvents) {
-					if (activationEvent === 'onStartupFinished') {
-						this._activateOneStartupFinished(desc, activationEvent);
+		this._extHostConfiguration.getConfigProvider().then((configProvider) => {
+			const shouldDeferActivation = configProvider.getConfiguration('extensions.experimental').get<boolean>('deferredStartupFinishedActivation');
+			for (const desc of this._myRegistry.getAllExtensionDescriptions()) {
+				if (desc.activationEvents) {
+					for (const activationEvent of desc.activationEvents) {
+						if (activationEvent === 'onStartupFinished') {
+							if (shouldDeferActivation) {
+								this._register(runWhenIdle(() => this._activateOneStartupFinished(desc, activationEvent)));
```

Comment:
I wonder if the we should let `_activeOne...` return its inner activation promise so that we can truly sequentialise this?

Reply:
Hm, interesting idea. Would that be in addition to or instead of using a timeout?

Reply:
I'd say in combination because otherwise you simply defer the activation of all extension "by 1"